### ISSUE TITLE
6.13. Conditional expressions: Grammar seems wrong

### ISSUE THREAD

**Issue by tvogel** [python/cpython#112515]
# Documentation

Under 6.13. Conditional expressions, the grammar is given as

```
conditional_expression ::=  or_test ["if" or_test "else" expression]
expression             ::=  conditional_expression | lambda_expr
```

I am surprised by the first `or_test` and would have expected it to be `expression` instead, i.e.:
```
conditional_expression ::=  expression ["if" or_test "else" expression]
expression             ::=  conditional_expression | lambda_expr
```

Is it a typo or am I missing something?

**Comment by skirpichev** [comment]
In the python.gram this looks like
https://github.com/python/cpython/blob/939fc6d6eab9b7ea8c244d513610dbdd556503a7/Grammar/python.gram#L675
where
https://github.com/python/cpython/blob/939fc6d6eab9b7ea8c244d513610dbdd556503a7/Grammar/python.gram#L709-L714

So, it seems docs are correct here.

**Comment by pan324** [comment]
`or_test` is basically an expression which does not support if-else or lambdas. This way `A if X else B if Y else Z` is always interpreted as `A if X else (B if Y else Z)` and `lambda x:x if 1 else -1` corresponds to `lambda x:(x if 1 else -1)`. Using `expression` twice would cause issues with ambiguity. (Though it would probably be not much of an issue since Python 3.9 due to using a PEG parser).

**Comment by StanFromIreland** [comment]
I agree with Sergey's analysis. I can also confirm that the grammar has not changed since.
